osquery/tables/networking/windows/routes.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <memory>
+#include <string>
+#include <windows.h>
+#include <winsock2.h>
+
+#include <wS2tcpip.h>
+#include <ws2ipdef.h>
+
+#include <iphlpapi.h>
+#include <mstcpip.h>
+
+#include <boost/algorithm/string/join.hpp>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+std::map<unsigned long, std::shared_ptr<IP_ADAPTER_INFO>>
+getAdapterAddressMapping() {
+  std::map<unsigned long, std::shared_ptr<IP_ADAPTER_INFO>> returnMapping;
+  auto dwBufLen = 0UL;
+  auto dwStatus = GetAdaptersInfo(NULL, &dwBufLen);
+
+  if (dwStatus == ERROR_BUFFER_OVERFLOW) {
```

Comment:
I agree it's odd, this is also what I found in the MSDN article w/r/t to the API call. [See step #3](https://msdn.microsoft.com/en-us/library/windows/desktop/aa366314(v=vs.85).aspx)